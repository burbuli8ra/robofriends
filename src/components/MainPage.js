import React, { Component } from 'react';
import Header from './Header';
import SearchBox from './SearchBox';
import Scroll from './Scroll';
import ErrorBoundary from './ErrorBoundary';
import CardList from './CardList';
import '../styles/MainPage.css';

class MainPage extends Component {
  componentDidMount() {
    this.props.onRobotsRequest();
  }

  filterRobots = () => {
    const {searchField, robots} = this.props;

    return robots.filter(robot => (
      robot.name.toLowerCase().includes(searchField.toLowerCase()))
    );
  };

  render() {
    const { onSearchChange, isPending } = this.props;

    return isPending ?
      <h1 className='f2 tc'>Loading...</h1> :
      (
        <div className='tc'>
          <Header />
          <SearchBox searchChange={onSearchChange} />
          <Scroll>
            <ErrorBoundary>
              <CardList robots={this.filterRobots()} />
            </ErrorBoundary>
          </Scroll>
        </div>
      );
  }
}

export default MainPage;
