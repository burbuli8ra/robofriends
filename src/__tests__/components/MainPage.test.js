import React from 'react';
import { shallow } from 'enzyme';
import MainPage from '../../components/MainPage';

let wrapper;

beforeEach(() => {
  const mockProps = {
    onRobotsRequest: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false
  };
  wrapper = shallow(<MainPage { ...mockProps } />);
});

describe('MainPage Component', () => {
  it('should render MainPage component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should filter robots correctly', () => {
    expect(wrapper.instance().filterRobots()).toEqual([]);
  });

  it('should render loading component', () => {
    const mockPropsPending = {
      onRobotsRequest: jest.fn(),
      robots: [],
      searchField: '',
      isPending: true
    };
    const wrapperPending = shallow(<MainPage { ...mockPropsPending } />);

    expect(wrapperPending).toMatchSnapshot();
  });

  it('should filter robots for John correctly', () => {
    const mockPropsJohn = {
      onRobotsRequest: jest.fn(),
      robots: [
        {
          id: 1,
          name: 'John Snow',
          username: 'John John',
          email: 'johnsnow@gmail.com'
        }
      ],
      searchField: 'john',
      isPending: false
    };

    const wrapperJohn = shallow(<MainPage { ...mockPropsJohn } />);

    expect(wrapperJohn.instance().filterRobots()).toEqual([{
      id: 1,
      name: 'John Snow',
      username: 'John John',
      email: 'johnsnow@gmail.com'
    }]);
  });

  it('should filter robots for \'a\' correctly', () => {
    const mockPropsA = {
      onRobotsRequest: jest.fn(),
      robots: [
        {
          id: 1,
          name: 'John Snow',
          username: 'John John',
          email: 'johnsnow@gmail.com'
        }
      ],
      searchField: 'a',
      isPending: false
    };

    const wrapperA = shallow(<MainPage { ...mockPropsA } />);

    expect(wrapperA.instance().filterRobots()).toEqual([]);
  });
});