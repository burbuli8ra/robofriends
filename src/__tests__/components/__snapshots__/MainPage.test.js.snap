// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Component should render MainPage component 1`] = `ShallowWrapper {}`;

exports[`MainPage Component should render loading component 1`] = `ShallowWrapper {}`;
